src/cbmc/cbmc_parse_options.cpp

```diff
@@ -430,6 +434,81 @@ void cbmc_parse_optionst::get_command_line_options(optionst &options)
 
 /*******************************************************************\
 
+Function: cbmc_parse_optionst::unwind_enum_static
+
+  Inputs: goto_functions and options
+
+ Outputs: side effect is adding <clinit> loops to unwindset
+
+ Purpose: unwind static initialization loops of Java enums as far as the enum
+          has elements
+
+\*******************************************************************/
+
+void cbmc_parse_optionst::unwind_enum_static(
+  const goto_functionst &goto_functions,
+  optionst &options)
+{
+  size_t unwind_max=0;
+  forall_goto_functions(f, goto_functions)
+  {
+    auto &p=f->second.body;
+    for(auto it : p.instructions)
+    {
+      // is this a loop?
+      if(it.is_backwards_goto())
+      {
+        size_t loop_id=it.loop_number;
+        const std::string java_clinit="<clinit>:()V";
+        const std::string &fname=id2string(it.function);
+        size_t class_prefix_length=fname.find_last_of('.');
+        assert(
+          class_prefix_length!=std::string::npos &&
+          "could not identify class name");
+        const std::string &classname=fname.substr(0, class_prefix_length);
+        const symbolt &class_symbol=symbol_table.lookup(classname);
+        const class_typet &class_type=to_class_type(class_symbol.type);
+        size_t unwinds=class_type.get_size_t(ID_java_enum_static_unwind);
+
+        unwind_max=std::max(unwind_max, unwinds);
+        if(fname.length()>java_clinit.length())
+        {
+          // extend unwindset with unwinds for <clinit> of enum
+          if(fname.compare(
+            fname.length()-java_clinit.length(),
+            java_clinit.length(),
+            java_clinit)==0 && unwinds>0)
+          {
+            status() << "unwinding enum " << classname
+                     << " loopID "
+                     << (id2string(it.function)+"."+std::to_string(loop_id)+" ")
+                     << std::to_string(unwinds) << " times"
+                     << eom;
+            const std::string &set=options.get_option("unwindset");
+            std::string newset;
+            if(set!="")
+              newset=",";
+            newset+=
+              id2string(it.function)+"."+std::to_string(loop_id)+":"+
+              std::to_string(unwinds);
+            options.set_option("unwindset", set+newset);
+          }
+        }
+      }
+    }
+  }
+  const std::string &vla_length=options.get_option("java-max-vla-length");
+  size_t max_vla_length=vla_length==""?0:unsafe_string2unsigned(vla_length);
```

Comment:
As you are using `unsafe_...` you don't actually need to have a `vla_length==""?0` guard as `0` will be the default return value.